src/library/treeitemmodel.cpp

```diff
@@ -164,22 +164,21 @@ TreeItem* TreeItemModel::setRootItem(std::unique_ptr<TreeItem> pRootItem) {
 
 /**
  * Before you can resize the data model dynamically by using 'insertRows' and 'removeRows'
- * make sure you have initialized
+ * make sure you have initialized.
  */
 void TreeItemModel::insertTreeItemRows(
-        QList<TreeItem*>& rows,
+        std::vector<std::unique_ptr<TreeItem>>&& rows,
```

Comment:
the heap allocations happens in the calling code. For instance in BasePlaylistFeature::constructChildModel Line 640

Reply:
Yes, but there are at least two locations where it could be avoided if we instead took a span instead of a rvalue-ref to a vector because the caller could just use an array and create a span from that.

https://github.com/mixxxdj/mixxx/pull/11230/files#diff-0e02fa72cc2bf68948f489976e3edf7634fa84b5f03b77e545593018565b81dbR218-R222

https://github.com/mixxxdj/mixxx/pull/11230/files#diff-16fa56a50091f69845e3552ebd062b6902ffe9ed849825809bab4b75106acc89R154-R156